File: Seccion 11 - Funciones en C++/Seccion_11_Python/10_ejercicio_08.py

cien = 0
cincuenta = 0
veinte = 0
diez = 0
cinco = 0
uno = 0

def cambio(valor):
    global cien, cincuenta, veinte, diez, cinco, uno

    cien = int(valor / 100)
    valor %= 100
    cincuenta = int(valor / 50)
    valor %= 50
    veinte = int(valor /20)
    valor %= 20
    diez = int(valor / 10)
    valor %= 10
    cinco = int(valor / 5)
    uno = valor % 5

valor = int(input('Digite la cantidad en dólares: '))

cambio(valor)

print('Cantidad de billetes a entregar como cambio: ')
print('Cien: {}'.format(cien))
print('Cincuenta: {}'.format(cincuenta))
print('Veinte: {}'.format(veinte))
print('Diez: {}'.format(diez))
print('Cinco: {}'.format(cinco))
print('Uno: {}'.format(uno))

File: Seccion 11 - Funciones en C++/Seccion_11_Dart/10_ejercicio_08.dart


import 'dart:io';

class Cambio {
  int cien = 0, cincuenta = 0, veinte = 0, diez = 0, cinco = 0, uno = 0;

  Cambio();
}

void cambio(int valor, Cambio exchange) {
  exchange.cien = (valor / 100).floor();
  valor %= 100;
  exchange.cincuenta = (valor / 50).floor();
  valor %= 50;
  exchange.veinte = (valor / 20).floor();
  valor %= 20;
  exchange.diez = (valor / 10).floor();
  valor %= 10;
  exchange.cinco = (valor / 5).floor();
  exchange.uno = valor % 5;
}

void main() {

  stdout.writeln('Digite la cantidad en dólares: ');
  int valor = int.parse(stdin.readLineSync());

  Cambio exchange = new Cambio();

  cambio(valor, exchange);

  print('Cantidad de billetes a entregar como cambio: ');
  print('Cien: ${exchange.cien}');
  print('Cincuenta: ${exchange.cincuenta}');
  print('Veinte: ${exchange.veinte}');
  print('Diez: ${exchange.diez}');
  print('Cinco: ${exchange.cinco}');
  print('Uno: ${exchange.uno}');

}